src/gt4py/next/program_processors/runners/dace_fieldview/gtir_dataflow.py

```diff
@@ -1309,3 +1324,45 @@ def visit_SymRef(self, node: gtir.SymRef) -> IteratorExpr | MemletExpr | SymbolE
         # if not in the lambda symbol map, this must be a symref to a builtin function
         assert param in gtir_python_codegen.MATH_BUILTINS_MAPPING
         return SymbolExpr(param, dace.string)
+
+    def apply(
+        self,
+        node: gtir.Lambda,
+        args: Sequence[IteratorExpr | MemletExpr | SymbolExpr],
+    ) -> tuple[list[DataflowInputEdge], DataflowOutputEdge]:
+        """
+        Entry point for this visitor class.
+
+        This visitor will translate a `Lambda` node into a dataflow graph to be
+        instantiated inside a map scope implementing the field operator.
+        However, this `apply()` method is responsible to recognize the usage of
+        the `Lambda` node, which can be either a let-statement or the stencil expression
+        in local view. The usage of a `Lambda` as let-statement corresponds to computing
+        some results and making them available inside the lambda scope, represented
+        as a nested SDFG. All let-statements, if any, are supposed to be encountered
+        before the stencil expression. In other words, the `Lambda` node representing
+        the stencil expression is always the innermost node.
+        Therefore, the lowering of let-statements results in recursive calls to
+        `apply()` until the stencil expression is found. At that point, it falls
+        back to the `visit()` function.
+        """
+
+        # lambda arguments are mapped to symbols defined in lambda scope.
+        prev_symbol_map = self.symbol_map
+        self.symbol_map = self.symbol_map.copy()
```

Comment:
```suggestion
        prev_symbol_map = self.symbol_map.copy()
```

Reply:
I was thinking to save the `self.symbol_map` object as `prev_symbol_map`, just to be on the safe side.

Reply:
In my view this is indicating that you have some big problems.